deeppavlov/models/kbqa/tree_parser.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from io import StringIO
+from typing import List, Tuple
+
+from udapi.block.read.conllu import Conllu
+from udapi.core.node import Node
+from ufal_udpipe import Model as udModel, Pipeline
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.models.component import Component
+from deeppavlov.core.models.serializable import Serializable
+
+
+def descendents(node, desc_list):
+    if len(node.children) > 0:
+        for child in node.children:
+            desc_list = descendents(child, desc_list)
+    desc_list.append(node.form)
+
+    return desc_list
+
+
+@register('tree_parser')
+class TreeParser(Component, Serializable):
+    """
+        This class parses the question using UDPipe to detect entity and relation
+    """
+
+    def __init__(self, load_path: str,
+                 udpipe_filename: str, **kwargs) -> None:
+        """
+        
+        Args:
+            load_path: path to file with UDPipe model
+            udpipe_filename: filename with UDPipe model
+            **kwargs
+        """
+        super().__init__(save_path=None, load_path=load_path)
+        self.udpipe_filename = udpipe_filename
+        self.udpipe_load_path = self.load_path / self.udpipe_filename
+        self.ud_model = udModel.load(str(self.udpipe_load_path))
+        self.full_ud_model = Pipeline(self.ud_model, "vertical", Pipeline.DEFAULT, Pipeline.DEFAULT, "conllu")
+
+    def load(self) -> None:
+        pass
+
+    def save(self) -> None:
+        pass
+
+    def __call__(self, q_tokens: List[str],
+                 *args, **kwargs) -> Tuple[str, str]:
```

Comment:
Components should expect batches of data and return batches of answers of the same size